streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1477,6 +1590,67 @@ public Object getStateLock() {
         return stateLock;
     }
 
+    public Map<String, KafkaFuture<Uuid>> consumerClientInstanceIds(final Duration timeout) {
```

Comment:
I have the same thread safety concerns here as I do with the global thread. Only they are even more complicated... :')